blocks/soapy/include/gnuradio-4.0/soapy/SoapyRaiiWrapper.hpp

```diff
@@ -0,0 +1,805 @@
+#ifndef SOAPYRAIIWRAPPER_HPP
+#define SOAPYRAIIWRAPPER_HPP
+
+#include <SoapySDR/Device.h> // using SoapySDR's C-API as intermediate interface to mitigate ABI-issues between stdlibc++ and libc++
+#include <SoapySDR/Formats.h>
+#include <SoapySDR/Modules.h>
+#include <SoapySDR/Version.h>
+
+namespace gr::blocks::soapy {
+
+using Range = SoapySDRRange;
+
+struct ArgInfo { // redeclare to be trivially constructable and ABI compatible
+    std::string key;
+    std::string value;
+    std::string name;
+    std::string description;
+    std::string units;
+    enum Type { BOOL, INT, FLOAT, STRING } type = STRING;
+    Range                    range{};
+    std::vector<std::string> options;
+    std::vector<std::string> optionNames;
+};
+
+namespace detail {
+
+/**
+ * @brief RAII Wrapper for SoapySDRKwargs
+ */
+class KwargsWrapper {
+    SoapySDRKwargs _cArgs;
+
+public:
+    KwargsWrapper(const SoapySDR::Kwargs& args) {
+        _cArgs.size = args.size();
+        _cArgs.keys = new char*[_cArgs.size];
+        _cArgs.vals = new char*[_cArgs.size];
+
+        std::size_t i = 0;
+        for (const auto& [key, value] : args) {
+            _cArgs.keys[i] = new char[key.size() + 1];
+            std::strcpy(_cArgs.keys[i], key.c_str());
+
+            _cArgs.vals[i] = new char[value.size() + 1];
+            std::strcpy(_cArgs.vals[i], value.c_str());
+
+            i++;
+        }
+    }
+
+    ~KwargsWrapper() {
+        for (size_t i = 0; i < _cArgs.size; ++i) {
+            delete[] _cArgs.keys[i];
+            delete[] _cArgs.vals[i];
+        }
+        delete[] _cArgs.keys;
+        delete[] _cArgs.vals;
```

Comment:
```suggestion
SoapySDRKwargs_clear(&_cArgs);
```
Can one use standard clear function from SoapySDR?

Reply:
done.